Configurations.md

```diff
@@ -2692,6 +2692,17 @@ By default this option is set as a percentage of [`max_width`](#max_width) provi
 
 See also [`max_width`](#max_width) and [`use_small_heuristics`](#use_small_heuristics)
 
+## `style_edition`
+
+Controls the edition of the [Rust Style Guide] to use for formatting ([RFC 3338])
+
+- **Default value**: `"2015"`
+- **Possible values**: `"2015"`, `"2018"`, `"2021"`, `"2024"` (unstable variant)
+- **Stable**: Yes
```

Comment:
I went back and forth on whether to insta-stabilize the option. I actually rolled the code back to make it and the CLI flag unstable but left this here as a note-to-self for us to discuss.

My inclination is to push this out as unstable, get some relevant high activity repos like r-l/rust switched over to it, etc. to make sure nothing goes bump. then we can stabilize it in the near future